src/box/journal.h

```diff
@@ -271,6 +275,7 @@ journal_write_submit(struct journal_entry *entry)
 	journal_queue_on_append(entry);
 	if (current_journal->write_async(current_journal, entry) != 0) {
 		journal_queue_on_complete(entry);
+		journal_queue_rollback();
```

Comment:
You could factor out common parts of `journal_queue_rollback()` and `journal_queue_wait`, where you roll back entries due to a fiber cancel, into a separate helper.

Up to you.

Reply:
Ok, as you wish.